src/client.ts

```diff
@@ -110,38 +97,35 @@ export class SubscriptionClient {
     this.client.close();
   }
 
-  public subscribe(options: SubscriptionOptions, handler: (error: Error[], result?: any) => void) {
-    const { query, variables, operationName, context } = options;
-
-    if (!query) {
-      throw new Error('Must provide `query` to subscribe.');
-    }
+  public query(options: RequestOptions): Promise<ExecutionResult> {
```

Comment:
Ok, I think if @Urigo prefers it this way, it's fine 🙂  We can rewrite it later.